lib/backend/helpers.go

```diff
@@ -61,3 +62,36 @@ func ReleaseLock(ctx context.Context, backend Backend, lockName string) error {
 	}
 	return nil
 }
+
+// jitter is used to allow multi-process reattempts to be randomly dispersed.
+var jitter = utils.NewJitter()
+
+// Swap emulates an upsert operation in that it will update an existing item,
+// otherwise create the item. Performs an additional operation to check if user exists first.
+// This allows us to differentiate between an update/create operation.
+func Swap(ctx context.Context, backend Backend, item Item) (*Item, error) {
+	const swapAttempts = 3
+	for i := 0; i < swapAttempts; i++ {
```

Comment:
This code needs an explanatory comment:
- why are multiple attempts needed?
- why are randomized sleeps needed?